reana_workflow_controller/rest/utils.py

```diff
@@ -538,3 +533,69 @@ def get_workflow_progress(workflow):
         "run_started_at": run_started_at,
         "run_finished_at": run_finished_at,
     }
+
+
+def use_paginate_args():
+    """Get and validate pagination arguments."""
+
+    def decorator(f):
+        @wraps(f)
+        def inner(*args, **kwargs):
+            try:
+                req = parser.parse(
+                    {
+                        "page": fields.Int(validate=validate.Range(min=1)),
+                        "size": fields.Int(validate=validate.Range(min=1)),
+                    },
+                    location="querystring",
+                    error_status_code=400,
+                )
+            # For validation errors, webargs raises an enhanced BadRequest
+            except BadRequest as err:
+                return jsonify({"message": err.data.get("messages")}), err.code
+
+            # Default if page is not specified
+            if not req.get("page"):
+                req["page"] = 1
+
+            if req.get("size"):
+                req.update(
+                    dict(
+                        from_idx=(req["page"] - 1) * req["size"],
+                        to_idx=req["page"] * req["size"],
+                        links=dict(
+                            prev={"page": req["page"] - 1},
+                            self={"page": req["page"]},
+                            next={"page": req["page"] + 1},
+                        ),
+                    )
+                )
+
+            def paginate(query_or_list):
```

Comment:
What do you think of adding a docstring to this method explaining that the query it might take is an SQLAlchemy query object? I think we might be confused with this function in the future if we don't explain it well.

Reply:
Yeah, definitely there's some technical debt in this part. I can add the docstring, no problem. I was also thinking about creating a `Pagination` class and populate instances of it to be returned in this function, instead of using a simple dictionary, what do you think? It can be another issue

Reply:
For me, it is fine as it is, as long as it is explained. We can create a separate issue for refactoring it and now is the best time as you have it fresh :)

Reply:
https://github.com/reanahub/reana-workflow-controller/issues/321